server/routes/bitbucket/bitbucket_selection.py

```
@@ -76,15 +163,78 @@ def save_workspace_selection(user_id):
+@bitbucket_selection_bp.route("/repo-metadata/generate", methods=["POST"])
+@require_permission("connectors", "write")
+def trigger_metadata_generation(user_id):
+    """Trigger LLM metadata generation for a specific Bitbucket repo. (Only for the regenerate button in the workspace browser)"""
+    try:
+        data = request.get_json()
+        repo_full_name = data.get("repo_full_name") if data else None
+        if not repo_full_name:
+            return jsonify({"error": "repo_full_name is required"}), 400
+
+        with db_pool.get_admin_connection() as conn:
+            with conn.cursor() as cur:
```

Ruff (0.15.15)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)